storage/innobase/gis/gis0sea.cc

```diff
@@ -278,9 +278,8 @@ rtr_pcur_getnext_from_path(
 				      ulint{RW_S_LATCH}, "");
 			static_assert(ulint{BTR_MODIFY_LEAF} ==
 				      ulint{RW_X_LATCH}, "");
-			rw_latch = (my_latch_mode | 4) == BTR_CONT_MODIFY_TREE
-				? RW_NO_LATCH
-				: rw_lock_type_t(my_latch_mode);
+			rw_latch = rw_lock_type_t(my_latch_mode
+						  & (RW_X_LATCH | RW_S_LATCH));
```

Comment:
I think this part of latching change should not be required. Please see the other comment.

Reply:
You are right, only the debug assertion seems to be at fault. I successfully tested the following:
```diff
diff --git a/storage/innobase/gis/gis0sea.cc b/storage/innobase/gis/gis0sea.cc
index fb4fcf7bc8d..9aedd4af69c 100644
--- a/storage/innobase/gis/gis0sea.cc
+++ b/storage/innobase/gis/gis0sea.cc
@@ -289,10 +289,6 @@ rtr_pcur_getnext_from_path(
 			mtr->rollback_to_savepoint(1);
 		}
 
-		ut_ad((my_latch_mode | 4) == BTR_CONT_MODIFY_TREE
-		      || !page_is_leaf(btr_cur_get_page(btr_cur))
-		      || !btr_cur->page_cur.block->page.lock.have_any());
-
 		const auto block_savepoint = mtr->get_savepoint();
 		block = buf_page_get_gen(
 			page_id_t(index->table->space_id,
```